parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -192,6 +194,11 @@ public WriteBuilder schema(Schema newSchema) {
       return this;
     }
 
+    public WriteBuilder variantShreddingFunc(BiFunction<Integer, String, Type> func) {
```

Comment:
This is a function passed to schema conversion. Variants (field ID and name) are passed to this function to determine the shredded type (`typed_value`). I'm using a callback function to avoid exposing a way to set the Parquet schema directly here.

Reply:
We should be documenting this, I don't think anyone will find this comment (see previous comment about codifying the interfaces)